### ISSUE TITLE
Fix false positives for nested rule-sets prefixed with comments in block-opening-brace-newline-after

### ISSUE THREAD

**Issue by iymeko** [stylelint/stylelint#3366]
**Describe the bug? (A clear and concise description of what the bug is)**

If a comment is placed before a descendant selector on the same line, `block-opening-brace-newline-after` shows an error. I'd really like for this not to happen because we follow Harry Roberts' idea of using [quasi-qualified selectors](https://cssguidelin.es/#quasi-qualified-selectors).

**Which rule, if any, is this issue related to?**

`block-opening-brace-newline-after`

**What SCSS is needed to reproduce this issue?**

```scss
.test1 {
    /*.test2*/.test3 {
        color: #f00;
    }
}
```
The issue also happens with one or more spaces between `/*.test2*/` and `.test3`, but not if there's a newline between `/*.test2*/` and `.test3`.

**What stylelint configuration is needed to reproduce this issue?**

```json
{
  "rules": {
    "block-opening-brace-newline-after": "always"
  }
}
```

**Which version of stylelint are you using?**

`9.1.3`

**How are you running stylelint: CLI, PostCSS plugin, Node API?**

`gulp-stylelint@7.0.0`
```javascript
gulp.task('lint', function () {
    var stylelint = require('gulp-stylelint');

    return gulp.src(sassFiles, { base: sassDir })
        .pipe(stylelint({
            reporters: [
                { formatter: 'string', console: true }
            ],
            failAfterError: false
        }));
});
```

**Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?**

Yes, it's related to SCSS nesting.

**What did you expect to happen?**

No warnings to be flagged.

**What actually happened (e.g. what warnings or errors you are getting)?**

```shell
Content/sass/modules/frontend/_search-main.scss     
 186:9   ×  Expected newline after "{"                                     block-opening-brace-newline-after
```

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

I could verify this bug.

For the example above our demo shows following error:

```
1:9 error Expected newline after "{" (block-opening-brace-newline-after)
```

**Comment by jeddy3** [comment]
@iymeko Thanks for the report. As this is labelled as "help wanted", please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).

**Comment by YozhikM** [comment]
WIP

**Comment by hudochenkov** [comment]
Done in #3383.
